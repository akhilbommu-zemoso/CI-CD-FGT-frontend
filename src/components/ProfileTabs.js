import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
 import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import GenralProfile from './GeneralProfile'
import BasicProfile from './BasicProfile'
import ProfileList from './ProfileList'
function TabPanel(props) {
    const { children, value, index, ...other } = props;
    return (
        <div
            role="tabpanel"
            hidden={value !== index}
            id={`simple-tabpanel-${index}`}
            aria-labelledby={`simple-tab-${index}`}
            {...other}
        >
            {value === index && (
                <Box p={3}>
                    <Typography>{children}</Typography>
                </Box>
            )}
        </div>
    );
}
TabPanel.propTypes = {
    children: PropTypes.node,
    index: PropTypes.any.isRequired,
    value: PropTypes.any.isRequired,
};
function a11yProps(index) {
    return {
        id: `simple-tab-${index}`,
        'aria-controls': `simple-tabpanel-${index}`,
    };
}
const useStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
        fontWeight: '16px',
        backgroundColor: theme.palette.background.paper,
    },
    red: {
        backgroundColor: 'white',
        boxShadow: "0px 0px 1px #9E9E9E"
    },
    color: {
        color: 'black',
        paddingTop: '20px',
    },
    text: {
        fontWeight: '16px',
        textTransform: 'none',
        fontWeight: 500,
        textAlign: 'center',
        '&:focus': {
            outline: 'none',
            border: 'none',
        },
    },
}));
export default function ProfileTabs(props) {
    const classes = useStyles();
    const [value, setValue] = React.useState(0);
    const handleChange = (event, newValue) => {
        setValue(newValue);
    };
    return (
        <div className={classes.root}>
            <AppBar position='static' className={classes.red}>
                <Tabs indicatorColor="primary"
                    textColor="primary" value={value} onChange={handleChange} className={classes.color} >
                    <Tab classes={{ root: classes.text }} label="Basic Profile" {...a11yProps(0)} />
                    <Tab className={classes.text} label="General" {...a11yProps(1)} />
                </Tabs>
            </AppBar>
            <TabPanel value={value} index={0}>
                {props.tab1}
            </TabPanel>
            <TabPanel value={value} index={1}>
               {props.tab2}
            </TabPanel>
        </div>
    );
}