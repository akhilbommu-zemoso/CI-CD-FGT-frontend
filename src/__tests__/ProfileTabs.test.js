import React, { Component } from 'react'
import { render, fireEvent, getAllByText } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import ProfileTabs from '../components/ProfileTabs'
it('should take a snapshot', () => {
    const { asFragment, getByTestId } = render(<Router><ProfileTabs /></Router>)
    expect(asFragment(<Router><ProfileTabs /></Router>)).toMatchSnapshot()
});

