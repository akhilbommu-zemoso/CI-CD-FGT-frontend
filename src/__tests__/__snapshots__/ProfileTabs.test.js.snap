// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-red-2 MuiPaper-elevation4"
    >
      <div
        class="MuiTabs-root makeStyles-color-3"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed"
          style="overflow: hidden;"
        >
          <div
            class="MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-controls="simple-tabpanel-0"
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root makeStyles-text-4 MuiTab-textColorPrimary Mui-selected"
              id="simple-tab-0"
              role="tab"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiTab-wrapper"
              >
                Basic Profile
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-controls="simple-tabpanel-1"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary makeStyles-text-4"
              id="simple-tab-1"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiTab-wrapper"
              >
                General
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="PrivateTabIndicator-root-68 PrivateTabIndicator-colorPrimary-69 MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </header>
    <div
      aria-labelledby="simple-tab-0"
      id="simple-tabpanel-0"
      role="tabpanel"
    >
      <div
        class="MuiBox-root MuiBox-root-73"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        />
      </div>
    </div>
    <div
      aria-labelledby="simple-tab-1"
      hidden=""
      id="simple-tabpanel-1"
      role="tabpanel"
    />
  </div>
</DocumentFragment>
`;
